cmd/cdi-uploadproxy/uploadproxy.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	. "kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/pkg/uploadproxy"
+)
+
+const (
+	// Default port that api listens on.
+	defaultPort = 8443
+
+	// Default address api listens on.
+	defaultHost = "0.0.0.0"
+)
+
+var (
+	configPath string
+	masterURL  string
+	verbose    string
+)
+
+func init() {
+	// flags
+	flag.StringVar(&configPath, "kubeconfig", os.Getenv("KUBECONFIG"), "(Optional) Overrides $KUBECONFIG")
+	flag.StringVar(&masterURL, "server", "", "(Optional) URL address of a remote api server.  Do not set for local clusters.")
+	flag.Parse()
+
+	// get the verbose level so it can be passed to the importer pod
+	defVerbose := fmt.Sprintf("%d", DEFAULT_VERBOSE) // note flag values are strings
+	verbose = defVerbose
+	// visit actual flags passed in and if passed check -v and set verbose
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == "v" {
+			verbose = f.Value.String()
+		}
+	})
+	if verbose == defVerbose {
+		glog.V(Vuser).Infof("Note: increase the -v level in the api deployment for more detailed logging, eg. -v=%d or -v=%d\n", Vadmin, Vdebug)
+	}
+}
+
+func main() {
+	defer glog.Flush()
+
+	cfg, err := clientcmd.BuildConfigFromFlags(masterURL, configPath)
+	if err != nil {
+		glog.Fatalf("Unable to get kube config: %v\n", errors.WithStack(err))
+	}
+	client, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		glog.Fatalf("Unable to get kube client: %v\n", errors.WithStack(err))
+	}
+	apiServerPublicKey, err := getAPIServerPublicKey()
+	if err != nil {
+		glog.Fatalf("Unable to get apiserver public key %v\n", errors.WithStack(err))
+	}
+
+	uploadProxy, err := uploadproxy.NewUploadProxy(defaultHost,
```

Comment:
Shouldn't we allow specifying a port and host here somewhere?  Maybe I'm missing where this happens but it appears we're just passing in the constants we set as defaults into these New functions.  If we're just forcing that then maybe the parameters should be eliminated from NewXxx and handled in the respective modules?